src/reactors/NonLinearDutchOrderReactor.sol

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {BaseReactor} from "./BaseReactor.sol";
+import {IPermit2} from "permit2/src/interfaces/IPermit2.sol";
+import {Permit2Lib} from "../lib/Permit2Lib.sol";
+import {ExclusivityLib} from "../lib/ExclusivityLib.sol";
+import {NonLinearDutchDecayLib} from "../lib/NonLinearDutchDecayLib.sol";
+import {
+    NonLinearDutchOrderLib,
+    NonLinearDutchOrder,
+    CosignerData,
+    NonLinearDutchOutput,
+    NonLinearDutchInput
+} from "../lib/NonLinearDutchOrderLib.sol";
+import {SignedOrder, ResolvedOrder} from "../base/ReactorStructs.sol";
+
+/// @notice Reactor for non-linear dutch orders
+/// @dev V2 orders must be cosigned by the specified cosigner to override timings and starting values
+/// @dev resolution behavior:
+/// - If cosignature is invalid or not from specified cosigner, revert
+/// - If inputAmount is 0, then use baseInput
+/// - If inputAmount is nonzero, then ensure it is less than specified baseInput and replace startAmount
+/// - For each outputAmount:
+///   - If amount is 0, then use baseOutput
+///   - If amount is nonzero, then ensure it is greater than specified baseOutput and replace startAmount
```

Comment:
is this comment still just the v2 one? seems it could use some extra nonlinear info

Reply:
Yup these are all still the same for non-linear orders. Just updated the header comment.